clang/lib/Sema/SemaLambda.cpp

```diff
@@ -1723,7 +1723,7 @@ FieldDecl *Sema::BuildCaptureField(RecordDecl *RD,
     // For SYCL compilations, save user specified names for
     // lambda capture.
     if (getLangOpts().SYCLIsDevice || getLangOpts().SYCLIsHost) {
-      StringRef CaptureName = Var->getName();
+      StringRef CaptureName = Var ? Var->getName() : "";
```

Comment:
So binding declarations are just null VarDecls?

Reply:
> So binding declarations are just null VarDecls?

Yeah, that feels weird to me too.
Also, they seem to have a name.

Reply:
> So binding declarations are just null VarDecls?

No.   Originally this routine had a VarDecl return type.   This was changed to return a ValueDecl as part of the support for lambdas capturing BindingDecls.  So now, in this routine, when we dynamic cast to VarDecl, there is a possibility that Var could be NULL.   That is the bad null-dereference that this PR is fixing.

Reply:
A ValueDecl is a NamedDecl. Is there a reason why we can't retain the name for this as well? The reason we retained names for captures is so that the metadata generated for openCL kernel arguments retained use specified names - which we are now not doing for this case.